lib/llm.ex

```diff
@@ -0,0 +1,21 @@
+defmodule LangChain.LLM do
+  @moduledoc """
+  A behaviour for LLM
+  """
+
+  alias LangChain.Message
+  alias LangChain.MessageDelta
+
+  @type call_response :: {:ok, Message.t() | [Message.t()]} | {:error, String.t()}
+
+  @callback call(
+              struct(),
+              String.t() | [Message.t()],
+              [LangChain.Function.t()],
+              nil | (Message.t() | MessageDelta.t() -> any())
+            ) :: call_response()
+
+  def call(%model{} = config, messages, functions, callback_fn) do
+    model.call(config, messages, functions, callback_fn)
```

Comment:
This actually looks like a poor man's protocol 😅 

May be it would fit better? and `LangChain.ChatModels.ChatOpenAI` would implement it as something like:
```elixir
defmodule LangChain.ChatModels.ChatOpenAI do
  use Ecto.Schema
  # ...
  embedded_schema do
    # ...
  end
  
  def new(..) do
    ...
  end
  
  # ...
  
  defimpl LLM do
    def call(openai, prompt, functions \\ [], callback_fn \\ nil)
    # ...
  end
end
```
?